.github/workflows/Automation Master Legacy Workflow.yml

```diff
@@ -279,40 +279,125 @@ jobs:
             exit(1)
         shell: python
 
+  artifact_creation:
+      name: Artifact Creation
+      runs-on: ubuntu-latest
+      steps:
+        # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+        - uses: actions/checkout@v4
+          with:
+            fetch-depth: 0
+  
+        - name: Initialize
+          run: |
+            echo "workspace" ${{ github.workspace }}
+            echo "ref name" ${{ inputs.referenceName }}
+            echo "run number" ${{ inputs.runNumber }}
+            echo "ref type" ${{ inputs.referenceType }}
+            echo "repository" ${{ inputs.repository }}
+  
+        - name: Find .sln file
+          id: findSlnFile        
+          run: |
+            echo solutionFilePath=$(find . -type f -name '*.sln') >> $GITHUB_OUTPUT
+          shell: bash   
+          
+          # TODO: Refactor this in the future to a single stage with a loop that adds all the sources you specify.
+        - name: Enable Skyline GitHub NuGet Registry
+          if: inputs.owner == 'SkylineCommunications'
+          run: |
+            $SOURCE_NAME="PrivateGitHubNugets"
+            $SOURCE_URL="https://nuget.pkg.github.com/SkylineCommunications/index.json"
+            
+            # Check if the source exists. If it does, update it.
+            if (dotnet nuget list source | Select-String -Pattern $SOURCE_NAME) {
+              Write-Host "Updating existing source $SOURCE_NAME."
+              dotnet nuget update source $SOURCE_NAME --source $SOURCE_URL --username USERNAME --password ${{ secrets.GITHUB_TOKEN }} --store-password-in-clear-text
```

Comment:
Won't work correctly on legacy style projects. This was probably copy pasted from the SDK one. The command needed here is: 'nuget sources update'